### ISSUE TITLE
netconvert didn't join junctions with traffic_light type and priority type

### ISSUE THREAD

**Issue by GitRKhanHub** [eclipse/sumo#13060]
I try to convert OSM data using commands

```
--geometry.remove,
--roundabouts.guess,
--ramps.guess,
--junctions.join,
--tls.guess-signals,
--tls.discard-simple,
--tls.join,
--tls.default-type actuated,
--output.original-names,
--output.street-names
```.

But as a result neconvert didn't join close junctions with traffic_light type and priority type.

![image](https://user-images.githubusercontent.com/47250609/233022527-d1f890ab-5d2d-4835-947d-27d5c57833f6.png)

I've tried to use such commands like

```
--tls.guess,
--tls.taz-nodes,
--tls.guess.joining,
--tls.join-dist 100,
--tls.guess-signals.dist 100,
--tls.guess-signals.slack
```, but they didn't help me.

Please advise what to do.

**SUMO-version:** 1.15.0

**operating system:** win10

**Comment by GitRKhanHub** [comment]
Thank you so much for your answer!

In my case joining process should be made in automatic mode, not manual join. Manual join in netedit works very well, but then I use netconvert commands in terminal - junctions don't join anyway. Don't see anything specific in netconvert logs, attach log text below

`Warning: No way found for reference '229960685' in relation '6474432'
Warning: No way found for reference '996149788' in relation '6474432'
Warning: No way found for reference '202374199' in relation '6474432'
Warning: Ignoring restriction relation '6474432' with unknown from-way.
Warning: Ignoring restriction relation '6474432' with unknown to-way.
Warning: Ignoring restriction relation '6474432' with unknown via.
Warning: No way found for reference '669215957' in relation '9308577'
Warning: Ignoring restriction relation '9308577' with unknown from-way.
Warning: No way found for reference '48003322' in relation '9537094'
Warning: Ignoring restriction relation '9537094' with unknown from-way.
Warning: direction of restriction relation could not be determined
Warning: Ignoring restriction relation '9537095'.
Warning: Ignoring restriction relation '11096550' with unknown from-way.
Warning: Ignoring restriction relation '11096550' with unknown to-way.
Warning: direction of restriction relation could not be determined
Warning: Ignoring restriction relation '11096552'.
Warning: PT line 'Автобус № 1: улица Олега Кошевого => СК "Янтарный"' in relation 13094869 seems to be split, only keeping first part.
Warning: PT line 'Автобус № 1: СК "Янтарный" => улица Олега Кошевого' in relation 13094870 seems to be split, only keeping first part.
Warning: PT line 'Автобус № 7: СО Победа => улица Павлика Морозова' in relation 13095628 seems to be split, only keeping first part.
Warning: PT line 'Автобус № 7: улица Павлика Морозова => СНТ Победа' in relation 13095629 seems to be split, only keeping first part.
Warning: PT line 'Автобус № 25: Северная гора => СНТ Медик' in relation 13103053 seems to be split, only keeping first part.
Warning: PT line in relation 13110569 with no stops ignored. Probably OSM file is incomplete.
Warning: PT line in relation 13110570 with no stops ignored. Probably OSM file is incomplete.
Warning: Could not assign stop '1237081947' to pt line '13094870' (closest edge '980510419', distance 765.30). Ignoring!
Warning: Removed invalid stop '1237081947' from line '13094870'.
Warning: Could not assign stop '1237081947' to pt line '13095628' (closest edge '980510419', distance 765.30). Ignoring!
Warning: Removed invalid stop '1237081947' from line '13095628'.
Warning: Could not assign stop '1237081946' to pt line '13098777' (closest edge '980510419', distance 308.46). Ignoring!
Warning: Removed invalid stop '1237081946' from line '13098777'.
Warning: Could not assign stop '1237081947' to pt line '13103053' (closest edge '980510419', distance 765.30). Ignoring!
Warning: Removed invalid stop '1237081947' from line '13103053'.
Warning: Could not assign stop '1237081947' to pt line '13106966' (closest edge '980510419', distance 765.30). Ignoring!
Warning: Removed invalid stop '1237081947' from line '13106966'.
Warning: Removing pt stop '1237081946' on non existing edge ''.
Warning: Removing pt stop '1237081947' on non existing edge ''.
Warning: Removing pt stop '1237081948' on non existing edge ''.
Warning: Removed 3 pt stops because they could not be assigned to the network
Warning: 11 total messages of type: Could not assign stop '%' to pt line '%' (closest edge '%', distance %). Ignoring!
Warning: 7 total messages of type: No way found for reference '%' in relation '%'
Warning: 16 total messages of type: PT line '%' in relation % seems to be split, only keeping first part.
Warning: 11 total messages of type: Removed invalid stop '%' from line '%'.
Success.`

I've already tried to join junctions after OSM converting. This is example of command line input
`netconvert -s "road_map.net.xml" --tls.set "2893352688" --tls.join --tls.join-dist 50 --tls.guess-signals --tls.guess-signals.dist 50 -o "res.xml"` It gives the same result - junctions aren't joined.

I attach archive with osm data and net file below

[forNetconvert.zip](https://github.com/eclipse/sumo/files/11279715/forNetconvert.zip)

**Comment by namdre** [comment]
they problem here is that osm node 2893352688 is tagged as pedestrian crossing and this information currently isn't used in the joining heuristic (only crossing=traffic_signals would have an impact).

**Comment by GitRKhanHub** [comment]
Thank you! Has **netconvert** got a command like `--junctions.join-exclude` but `--junctions.join-include`, in which we could set a list of junctions we would to join?

**Comment by behrisch** [comment]
No, but you can define them in the plain XML see https://sumo.dlr.de/docs/Networks/PlainXML.html#joining_nodes

**Comment by namdre** [comment]
not quite since it's awkward to specify lists of lists on the command line (there might be multiple clusters to specify). Instead you can create a patch file to hold the explicit join directives and pass it on the command line with **-n patch.xml** when running netconvert. See https://sumo.dlr.de/docs/Networks/PlainXML.html#specifying_and_excluding_explicit_joins

**Comment by GitRKhanHub** [comment]
@behrisch , @namdre, plain XML worked! Thank you very much!
